setup.py

```diff
@@ -83,21 +82,19 @@ def read(f):
     return (here / f).read_text('utf-8').strip()
 
 
-class PyTest(TestCommand):
-    user_options = []
+NEEDS_PYTEST = {'pytest', 'test'}.intersection(sys.argv)
+pytest_runner = ['pytest-runner'] if NEEDS_PYTEST else []
 
-    def run(self):
-        import subprocess
-        errno = subprocess.call([sys.executable, '-m', 'pytest', 'tests'])
-        raise SystemExit(errno)
-
-
-tests_require = install_requires + ['pytest', 'gunicorn',
-                                    'pytest-timeout', 'async-generator']
+tests_require = ['pytest', 'gunicorn',
+                 'pytest-timeout', 'async-generator']
 
+name = 'aiohttp'
```

Comment:
I don't like `name` substitution.
A plain constant is more obvious than string-formatted, isn't it?

Reply:
I don't want to create code diversity, which often leads to forgetting to update some of places. Also, I use this approach in lots other distributions including multidict, which proved to be useful.

Reply:
I've approved it for multidict but now I'm inclining to revert bare strings back.
It is written once but read many times.
Every time when I see the line I should remember what *name* is.
You've convinced me that we need overcomplicated travis config. Perhaps it is true, I'm not Travis expert.
But for `setup.py` I pretty sure that the simplest approach is the best.

Reply:
I'd agree if we were using declarative `setup.cfg`, but this is code here.

Reply:
By the way I have an 80-line shim snippet for supporting newer setuptools, which would allow us to move a vast of meta/options to `setup.cfg`

Reply:
Say again I prefer constants to string substitutions in `setup.py`

Reply:
ok, as you wish. but how about moving that to `setup.cfg`?

Reply:
If you want to migrate -- I'm ok probably but want to see how it looks like first.
Could you point on another project with `setup.cfg` integration?

Reply:
AFAIR I only tried this in commercial ones, but I also tested out a shim, which makes it possible to support this in old versions of setuptools as well.

Reply:
Here's how it would've looked in ansible if we accepted the patch: https://github.com/ansible/ansible/pull/38413/files#diff-380c6a8ebbbce17d55d50ef17d3cf906R1

Reply:
+1 to move all to setup.cfg

Reply:
I know how to do it, it's just not going to be in the same PR.

Reply:
Also, there's certain complications related to this change. So let's not rush.